File: Code/07_Pipes_Sinais_Alarmes/Ex6.c
#include <errno.h>
#include <unistd.h>
#include <stdio.h>
#include <stdlib.h>
#include <signal.h>


void tratamento_SIGUSR1()
{
	printf("Processo %d recebeu sinal SIGUSR1... ele vai parar agora.\n", getpid());
	exit(1);
}

int main()
{
	int pid_filho;
	signal(SIGUSR1, tratamento_SIGUSR1);
	printf("Processo pai   [%d] vai criar o filho e dormir por 1 segundo.\n", getpid());
	if((pid_filho=fork())==0) 
	{
		printf("Processo filho [%d] vai entrar num loop infinito.\n", getpid());
		while(1);
	}
	else
	{
		sleep(1);
		printf("Processo %d vai enviar o sinal SIGUSR1 para o processo %d\n", getpid(), pid_filho);
		kill(pid_filho,SIGUSR1);
		printf("Processo %d vai dormir por 1 segundo.\n", getpid());
		sleep(1);
	}
	printf("Processo %d encerrando.\n", getpid());
	exit(0); 
}

File: 2_Programação_sistemas_operacionais/2.4_Pipes_Sinais_Alarmes/Python/Ex6.py
import os, signal, time

def tratamento_SIGUSR1(signum, frame):
	print("Processo %d recebeu sinal SIGUSR1... ele vai parar agora." % os.getpid())
	exit(1)

signal.signal(signal.SIGUSR1, tratamento_SIGUSR1);

pid_filho = os.fork()
if pid_filho == 0:
	print("Processo filho [%d] vai entrar num loop infinito." % os.getpid())
	while True:
		pass
else:
	time.sleep(1)
	print("Processo %d vai enviar o sinal SIGUSR1 para o processo %d" % (os.getpid(), pid_filho))
	os.kill(pid_filho, signal.SIGUSR1)
	print("Processo %d vai dormir por 1 segundo." % os.getpid())
	time.sleep(1)
print("Processo %d encerrando." % os.getpid())